test/support/rabbit.ts

```diff
@@ -117,6 +133,25 @@ export class Rabbit {
     return resp.body.map((p) => p.consumer_tag)
   }
 
+  async returnSingleConsumerCredits(): Promise<number> {
+    const allConsumersResp = await got.get<RabbitConsumersResponse[]>(`http://localhost:15672/api/consumers`, {
+      username: "rabbit",
+      password: "rabbit",
+      responseType: "json",
+    })
+    const consumerChannelDetails = allConsumersResp.body.map((d) => d.channel_details)
+    const connectionName = consumerChannelDetails[0].connection_name
+    const resp = await got.get<RabbitConnectionDetails[]>(
+      `http://localhost:15672/api/stream/connections/%2F/${connectionName}/consumers`,
+      {
+        username: "rabbit",
+        password: "rabbit",
+        responseType: "json",
+      }
+    )
+    return resp.body[0].credits
+  }
```

Comment:
I suggest returning an array for every `consumerChannelDetails`, and then in the test, you get the first. In this way, this method is more generic. Clearly, you have also to change the name to `returnConsumerCredits` or, better return all the information you have and select the first item and the credit in the test.